src/plugins/explore/public/components/data_table/explore_data_table.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import {
+  DEFAULT_COLUMNS_SETTING,
+  DOC_HIDE_TIME_COLUMN_SETTING,
+  MODIFY_COLUMNS_ON_SWITCH,
+  SAMPLE_SIZE_SETTING,
+} from '../../../common';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { IndexPatternField, opensearchFilters, UI_SETTINGS } from '../../../../data/public';
+import { DocViewFilterFn, OpenSearchSearchHit } from '../../types/doc_views_types';
+import { DataTable } from './data_table';
+import {
+  addColumn,
+  removeColumn,
+  useDispatch,
+  useSelector,
+} from '../../application/legacy/discover/application/utils/state_management';
+import { DiscoverViewServices } from '../../application/legacy/discover/build_services';
+import { useDiscoverContext } from '../../application/legacy/discover/application/view_components/context';
+import { popularizeField } from '../../application/legacy/discover/application/helpers/popularize_field';
+import { buildColumns } from '../../application/legacy/discover/application/utils/columns';
+import { filterColumns } from '../../application/legacy/discover/application/view_components/utils/filter_columns';
+import { getLegacyDisplayedColumns } from '../../helpers/data_table_helper';
+import { getDocViewsRegistry } from '../../application/legacy/discover/opensearch_dashboards_services';
+
+interface Props {
+  rows?: Array<OpenSearchSearchHit<Record<string, any>>>;
+  scrollToTop?: () => void;
+}
+
+export const ExploreDataTable = ({ rows, scrollToTop }: Props) => {
+  const { services } = useOpenSearchDashboards<DiscoverViewServices>();
+  const {
+    uiSettings,
+    data: {
+      query: { filterManager },
+    },
+    capabilities,
+    indexPatterns,
+  } = services;
+
+  const { indexPattern, savedSearch } = useDiscoverContext();
+
+  const { columns } = useSelector((state) => {
```

Comment:
how can visualization get these columns? is it stored under `state.visualization.columns`?